engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobFailProcessor.java

```diff
@@ -82,12 +91,26 @@ public void afterAccept(
   }
 
   private void acceptCommand(
-      final TypedRecord<JobRecord> command, final CommandControl<JobRecord> commandControl) {
+      final TypedRecord<JobRecord> command,
+      final CommandControl<JobRecord> commandControl,
+      final Consumer<SideEffectProducer> sideEffect) {
     final long key = command.getKey();
     final JobRecord failedJob = jobState.getJob(key);
-    failedJob.setRetries(command.getValue().getRetries());
+    final var retries = command.getValue().getRetries();
+    final var retryBackOff = command.getValue().getRetryBackOff();
+    failedJob.setRetries(retries);
     failedJob.setErrorMessage(command.getValue().getErrorMessageBuffer());
+    failedJob.setRetryBackoff(retryBackOff);
+    final long receivedTime = command.getTimestamp();
+    failedJob.setRecurringTime(receivedTime + retryBackOff);
```

Comment:
:x: We should set the `recurringTime` only if `retries > 0 && retryBackOff > 0`.

It doesn't break anything but it feels a bit inconsistent to set `recurringTime` if `retries == 0` or `retryBackOff == 0`.